mitodl/open-discussions
# Basic field edit form (add/remove/sort lists)

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #3677

#### What's this PR do?
Adds a form for specifying the lists and featured list for a field channel.

#### How should this be manually tested?
- Adjust the following in your .env file:
   ```
   OPEN_DISCUSSIONS_HOSTNAME=od.odl.local
   ENABLE_INFINITE_CORRIDOR=True
   ```
- Go to `../learn/`, and use the bookmark icons on items to create new public lists and add courses to each of them (make sure you choose "Public" for privacy level when creating new lists).
- Create some fields if you don't have any:
  - Go to `/api/v0/fields/` and post something like this JSON to create one:
    ```
    {
       "name": "physics",
       "title": "Physics"
    }
    ```  
- Go to `../infinite/fields/<field_name>/manage/`
- You should see a form with "Featured List" and "Lists" inputs.  If your field already has values for these, they should be visible here.
- Use the inputs to change/add/remove/reorder lists.  Start typing in each text input field and check that the available options gets narrowed down to those that match your text.
- Submit the form, you should end up back to the field's display page and your changes should be reflected there.


#### Any background context you want to provide?
The unit testing coverage is not as good as it could be, in particular I skipped testing the reordering of components via dragging because I couldn't seem to emulate that with RTL.

#### Screenshots (if appropriate)
<img width="882" alt="Screen Shot 2022-08-10 at 7 35 38 AM" src="https://user-images.githubusercontent.com/187676/183891699-35ac24be-abf4-4d50-a43d-05168ba91720.png">
